server/src/main/java/com/paypal/selion/utils/SauceConfigReader.java

```diff
@@ -36,20 +36,33 @@
 
     private static final SeLionGridLogger LOGGER = SeLionGridLogger.getLogger(SauceConfigReader.class);
 
+    /** Required. Credentials to connect to sauce labs. */
     public static final String AUTHENTICATION_KEY = "authenticationKey";
+    /** Required. REST endpoint to communicate with sauce labs. */
     public static final String SAUCE_URL = "sauceURL";
+    /** Optional. Connection time out for communicating with sauce labs api. Defaults to {@link #DEFAULT_TIMEOUT} */
     public static final String SAUCE_TIMEOUT = "sauceTimeout";
+    /** Optional. Connection retry for communicating with sauce labs api. Defaults to {@link #DEFAULT_RETRY_COUNT} */
     public static final String SAUCE_RETRY = "sauceRetries";
+    /** Optional. Default parent tunnel to use. Defaults to {@link #DEFAULT_TUNNEL} */
+    public static final String PARENT_TUNNEL = "parentTunnel";
+    /** Optional. Default tunnel to use. Defauls to {@link #DEFAULT_TUNNEL} */
+    public static final String TUNNEL_IDENTIFIER = "tunnelIdentifier";
+    /** Optional. Whether a user must provide their own sauce labs api credentials. Defaults to false */
+    public static final String REQUIRE_USER_CREDENTIALS = "requireUserCredentials";
 
     // Default of 10 seconds for connection & read
-    private static final int DEFAULT_TIMEOUT = 10 * 1000;
+    public static final int DEFAULT_TIMEOUT = 10 * 1000;
     // Number of retries before giving up on sauce
-    private static final int DEFAULT_RETRY_COUNT = 2;
+    public static final int DEFAULT_RETRY_COUNT = 2;
+    // No tunnel
+    public static final String DEFAULT_TUNNEL = "";
 
     private String authKey;
     private String sauceURL;
-    private String url;
-    private String userName;
+    private String defaultParentTunnel = DEFAULT_TUNNEL;
+    private String defaultTunnelIdentifier = DEFAULT_TUNNEL;
+    private boolean requireUserCredentials = false;
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid using redundant field initializer for 'requireUserCredentials'](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2417319208/issues/source?bid=2928425&fileBranchId=3233060#l65)